src/App.tsx

```diff
@@ -1,26 +1,279 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useState } from 'react';
+import { createTodo, deleteTodo, getTodos, updateTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { TodoItem } from './TodoItem';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Filter>(Filter.all);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processingIds, setProcessingIds] = useState<number[]>([]);
+  const [editId, setEditId] = useState<number | null>(null);
+
+  const [title, setTitle] = useState('');
+  const [editTitle, setEditTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const editInputRef = React.useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    if (!tempTodo) {
+      inputRef.current?.focus();
+    }
+  }, [tempTodo]);
+
+  useEffect(() => {
+    if (editId) {
+      editInputRef.current?.focus();
+    }
+  }, [editId]);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  let visibleTodos = todos;
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const normalizedTitle = title.trim();
+
+    if (!normalizedTitle) {
+      setErrorMessage('Title should not be empty');
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      title: normalizedTitle,
+      completed: false,
+      userId: 0,
+    });
+
+    setErrorMessage('');
+
+    createTodo(normalizedTitle)
+      .then(todo => {
+        setTodos((prevTodos: Todo[]) => [...prevTodos, todo]);
+        setTitle('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+      })
+      .finally(() => setTempTodo(null));
+  };
+
+  const handleSubmitEdit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const normalizedTitle = editTitle.trim();
+
+    if (!normalizedTitle) {
```

Comment:
you should delete it in case input is empty:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/86923089/749d487d-5059-4746-9bf6-ad999803f729)

Check working example (there is a placeholder)